library/src/main/java/org/apache/apex/malhar/lib/join/AbstractManagedStateInnerJoinOperator.java

```diff
@@ -90,17 +113,36 @@ public void createStores()
    * @param tuple given tuple
    * @param isStream1Data Specifies whether the given tuple belongs to stream1 or not.
    */
+  @SuppressWarnings("unchecked")
   @Override
   protected void processTuple(T tuple, boolean isStream1Data)
   {
     Spillable.SpillableListMultimap<K,T> store = isStream1Data ? stream1Data : stream2Data;
     K key = extractKey(tuple,isStream1Data);
-    long timeBucket = extractTime(tuple,isStream1Data);
-    if (!((ManagedTimeStateMultiValue)store).put(key, tuple,timeBucket)) {
+    //long timeBucket = extractTime(tuple,isStream1Data);
+    if (!store.put(key, tuple)) {
```

Comment:
Yes you are right. Before inserting, need to check whether the tuple is expired or not. For this check, I added isExpiredTuple() API.